hash/f1600x4/asm/go.mod

```diff
@@ -1,4 +1,4 @@
-module github.com/cloudflare/circl/sign/dilithium/internal/f1600x4/asm
+module github.com/cloudflare/circl/hash/f1600x4/asm
```

Comment:
Relocate `asm` folder to `internal/asm`.

Reply:
Done.